python/instrumentation/openinference-instrumentation-langchain/src/openinference/instrumentation/langchain/_tracer.py

```diff
@@ -648,18 +688,26 @@ def _extract_message_role(message_data: Optional[Mapping[str, Any]]) -> Iterator
     2. Try extracting from type field
     3. Try inferring from message context (tool_calls, tool_call_id, etc.)
     4. If all fail, log warning and skip role (span continues without role attribute)
+    
+    Special handling:
+    - RemoveMessage intentionally has no role and will not trigger a warning
     """
     if not message_data:
         return
     assert hasattr(message_data, "get"), f"expected Mapping, found {type(message_data)}"
 
     role = None
+    is_remove_message = False
 
     # Strategy 1: Try the standard id field approach
     id_ = message_data.get("id")
     if id_ is not None and isinstance(id_, List) and len(id_) > 0:
         try:
             message_class_name = id_[-1]
+            if message_class_name.startswith("RemoveMessage"):
+                is_remove_message = True
+                logger.debug("Encountered RemoveMessage - no role attribute needed")
+                return  # RemoveMessage doesn't need a role, exit early
```

Comment:
early return means that `is_remove_message` is never used, so the variable is superfluous
```suggestion
                logger.debug("Encountered RemoveMessage - no role attribute needed")
                return  # RemoveMessage doesn't need a role, exit early
```